clar2wasm/src/lib.rs

```diff
@@ -102,11 +104,19 @@ pub fn compile(
         }),
         Err(e) => {
             diagnostics.push(Diagnostic::err(&e));
-            Err(CompileError::Generic {
-                ast,
-                diagnostics,
-                cost_tracker: Box::new(contract_analysis.cost_track.take().unwrap()),
-            })
+            if let Some(inner_cost_tracker) = contract_analysis.cost_track.take() {
+                Err(CompileError::Generic {
+                    ast,
+                    diagnostics,
+                    cost_tracker: Box::new(inner_cost_tracker),
+                })
+            } else {
+                Err(CompileError::Generic {
+                    ast,
+                    diagnostics,
+                    cost_tracker: Box::new(cost_tracker),
+                })
+            }
```

Comment:
I think it's okay to use an `expect` in this case, since that is an unrecoverable error that would be caught in any run of the code. Let's revert to the original, but switching the `unwrap` to an `expect`, and then add an `allow` attribute for it.
```rust
            Err(CompileError::Generic {
                ast,
                diagnostics,
                cost_tracker: Box::new(
                    contract_analysis
                        .cost_track
                        .take()
                        .expect("Failed to take cost tracker from contract analysis"),
                ),
            })
```

Reply:
Alright. Fixed